src/control/controllers/PidController.cpp

```diff
@@ -0,0 +1,225 @@
+//
+// Created by mrlukasbos on 29-3-19.
+//
+
+#include <control/controllers/PidController.h>
+#include "utilities/Constants.h"
+
+//**********************************
+//Constructor functions
+//**********************************
+PidController::PidController(double p, double i, double d) : PidController(p,i,d,0){}
+
+PidController::PidController(double p, double i, double d, double f){
+    maxIOutput=0;
+    maxError=0;
+    errorSum=0;
+    maxOutput=0;
+    minOutput=0;
+    setpoint=0;
+    lastActual=0;
+    firstRun=true;
+    reversed=false;
+    outputRampRate=0;
+    lastOutput=0;
+    outputFilter=0;
+    setpointRange=0;
+
+    P=p; I=i; D=d; F=f;
```

Comment:
According to C++ policies you should never have multiple code lines at the same line, after a ; there should be a new line.